crates/doctor/src/manifest/trigger.rs

```diff
@@ -0,0 +1,236 @@
+use anyhow::{bail, ensure, Context, Result};
+use async_trait::async_trait;
+use toml::Value;
+use toml_edit::{Document, InlineTable, Item, Table};
+
+use crate::{Diagnose, Diagnosis, PatientApp, Treatment};
+
+use super::ManifestTreatment;
+
+/// TriggerDiagnosis detects problems with app trigger config.
+#[derive(Debug)]
+pub enum TriggerDiagnosis {
+    /// Missing app trigger section
+    MissingAppTrigger,
+    /// Invalid app trigger config
+    InvalidAppTrigger(&'static str),
+    /// HTTP trigger missing base field
+    HttpAppTriggerMissingBase,
+    /// HTTP component trigger missing route field
+    HttpComponentTriggerMissingRoute(String, bool),
+    /// Invalid HTTP component trigger config
+    InvalidHttpComponentTrigger(String, &'static str),
+}
+
+impl TriggerDiagnosis {
+    fn diagnose_app_trigger(config: Option<&Value>) -> Option<Self> {
+        let Some(config) = config else {
+            return Some(Self::MissingAppTrigger);
+        };
+        let Some(config) = config.as_table() else {
```

Comment:
I'm a bit worried that we're going to have to duplicate a lot of logic here and with the manifest parser...

Reply:
I think its inevitable for this kind of check unless we want to break the manifest parser up into pretty fine-grained stages with different levels of validation.